ctapipe/io/dstio.py

```diff
@@ -0,0 +1,266 @@
+                # data each time (right now it's just deleted and garbage
+                # collected)
+
+                data.r0.tel.clear()
+                data.r1.tel.clear()
+                data.dl0.tel.clear()
+                data.dl1.tel.clear()
+                data.mc.tel.clear()  # clear the previous telescopes
+
+                _fill_instrument_info(data, pyhessio)
+
+                for tel_id in data.r0.tels_with_data:
+
+                    # event.mc.tel[tel_id] = MCCameraContainer()
+
+                    data.mc.tel[tel_id].dc_to_pe \
+                        = pyhessio.get_calibration(tel_id)
+                    data.mc.tel[tel_id].pedestal \
+                        = pyhessio.get_pedestal(tel_id)
+
+                    calibrator = DstioR1Calibrator.calibrate_read
+                    image = calibrator(pyhessio.get_adc_sum(tel_id),
+                                       data.mc.tel[tel_id].pedestal,
+                                       data.mc.tel[tel_id].dc_to_pe)
+                    data.dl1.tel[tel_id].image = image
+                    # TODO: correctly obtain peakpos from the dst files,
+                    # requires changes to pyhessio
+                    data.dl1.tel[tel_id].peakpos = np.zeros(image.shape)
+                    data.mc.tel[tel_id].reference_pulse_shape = \
+                        pyhessio.get_ref_shapes(tel_id)
+
+                    nsamples = pyhessio.get_event_num_samples(tel_id)
+                    if nsamples <= 0:
+                        nsamples = 1
+                    data.r0.tel[tel_id].num_samples = nsamples
+
+                    # load the data per telescope/pixel
+                    hessio_mc_npe = pyhessio.get_mc_number_photon_electron
+                    data.mc.tel[tel_id].photo_electron_image \
+                        = hessio_mc_npe(telescope_id=tel_id)
+                    data.mc.tel[tel_id].meta['refstep'] = \
+                        pyhessio.get_ref_step(tel_id)
+                    data.mc.tel[tel_id].time_slice = \
+                        pyhessio.get_time_slice(tel_id)
+                    data.mc.tel[tel_id].azimuth_raw = \
+                        pyhessio.get_azimuth_raw(tel_id)
+                    data.mc.tel[tel_id].altitude_raw = \
+                        pyhessio.get_altitude_raw(tel_id)
+                    data.mc.tel[tel_id].azimuth_cor = \
+                        pyhessio.get_azimuth_cor(tel_id)
+                    data.mc.tel[tel_id].altitude_cor = \
+                        pyhessio.get_altitude_cor(tel_id)
+                yield data
+                counter += 1
+
+                if max_events is not None and counter >= max_events:
+                    pyhessio.close_file()
+                    return
+    except HessioError:
+        raise RuntimeError("dstio_event_source failed to open '{}'"
+                           .format(url))
+
+
+def _fill_instrument_info(data, pyhessio):
```

Comment:
also, now that code has changed to fill in the SubarrayDescription class, so this code no longer works. It's better to just do `from hessio.py import _fill_instrument_info` and use it